diff -ru xmms-1.2.10-orig/General/ir/ir.h xmms-1.2.10/General/ir/ir.h
--- xmms-1.2.10-orig/General/ir/ir.h	2001-03-05 15:17:44.000000000 +0200
+++ xmms-1.2.10/General/ir/ir.h	2006-04-08 11:42:02.000000000 +0300
@@ -49,8 +49,6 @@
 }
 irConfig;
 
-extern pthread_t irapp_thread;
-extern gboolean keepGoing;
 extern irConfig ircfg;
 extern gboolean irconf_is_going;
 
diff -ru xmms-1.2.10-orig/Output/esd/esdout.h xmms-1.2.10/Output/esd/esdout.h
--- xmms-1.2.10-orig/Output/esd/esdout.h	2001-12-03 04:38:42.000000000 +0200
+++ xmms-1.2.10/Output/esd/esdout.h	2006-04-08 11:42:47.000000000 +0300
@@ -58,6 +58,7 @@
 ESDConfig;
 
 extern ESDConfig esd_cfg;
+extern void esdout_reset_playerid(void);
 
 void esdout_init(void);
 void esdout_about(void);
