--- wmclock.c-orig	2008-11-16 18:23:07.185633821 +0100
+++ wmclock.c		2008-11-16 18:23:25.482650412 +0100
@@ -828,7 +828,14 @@
 #endif /* !ONLY_SHAPED_WINDOW */
    for (i = 0; i < NUM_TIME_POSITIONS; i++)
     {
-       xPos[i] += enable12HourClock ? timePos24[i] : timePos12[i];
+      if (enable12HourClock && (!enableYearDisplay))
+       {
+         xPos[i] += timePos24[i];
+       }
+      else
+       {
+         xPos[i] += timePos12[i];
+       }
     }
    
    /* Open the display */
--- wmclock.c-orig	2000-03-08 08:31:33.000000000 +0100
+++ wmclock.c		2007-10-03 22:42:36.000000000 +0200
@@ -470,11 +470,11 @@
    XCopyArea(dpy, led.pixmap, visible.pixmap, normalGC,
 	     digitXOffset , digitYOffset, LED_NUM_WIDTH, LED_NUM_HEIGHT,
 	     xPos[DIGIT_1_X_POS], yPos[DIGIT_Y_POS]);
-   digitXOffset = LED_NUM_WIDTH * (year % 1000);
+   digitXOffset = LED_NUM_WIDTH * (year % 1000 / 100);
    XCopyArea(dpy, led.pixmap, visible.pixmap, normalGC,
 	     digitXOffset , digitYOffset, LED_NUM_WIDTH, LED_NUM_HEIGHT,
 	     xPos[DIGIT_2_X_POS], yPos[DIGIT_Y_POS]);
-   digitXOffset = LED_NUM_WIDTH * (year % 100);
+   digitXOffset = LED_NUM_WIDTH * (year % 100 / 10);
    XCopyArea(dpy, led.pixmap, visible.pixmap, normalGC,
 	     digitXOffset , digitYOffset, LED_NUM_WIDTH, LED_NUM_HEIGHT,
 	     xPos[DIGIT_3_X_POS], yPos[DIGIT_Y_POS]);
