--- interceptor.c.orig	2005-05-11 18:48:42.000000000 +0200
+++ interceptor.c	2005-05-11 18:49:08.000000000 +0200
@@ -236,6 +236,24 @@
     dev_kfree_skb(skb);
     return 0;
 }
+
+static int
+inline supported_device(struct net_device* dev)
+{
+    int rc=0;
+
+    if(dev->type == ARPHRD_ETHER)
+    {
+        rc=1;
+    }
+    else if(dev->type == ARPHRD_PPP)
+    {
+        rc=1;
+    }
+
+    return rc;
+}
+
 static int
 add_netdev(struct net_device *dev)
 {
@@ -476,23 +494,6 @@
     s->rc = 0;
 }
 
-static int
-inline supported_device(struct net_device* dev)
-{
-    int rc=0;
-
-    if(dev->type == ARPHRD_ETHER)
-    {
-        rc=1;
-    }
-    else if(dev->type == ARPHRD_PPP)
-    {
-        rc=1;
-    }
-
-    return rc;
-}
-
 
 static BINDING *
 getbindingbydev(struct net_device *dev)
@@ -604,7 +605,7 @@
     if (skb->ip_summed == CHECKSUM_HW)
     {
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,7)
-       if (skb_checksum_help(&skb,1))
+       if (skb_checksum_help(skb,1))
        {
            dev_kfree_skb(skb);
            skb = NULL;
@@ -729,7 +730,7 @@
     if (skb->ip_summed == CHECKSUM_HW)
     {
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,7)
-       if (skb_checksum_help(&skb,0))
+       if (skb_checksum_help(skb,0))
 #else
        if ((skb = skb_checksum_help(skb)) == NULL)
 #endif //LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,7)
